src/thd_initdblk.c

```diff
@@ -722,6 +722,23 @@ int THD_WarpData_From_3dWarpDrive(THD_3dim_dataset *dset, ATR_float *atr_flt)
                 set "attr_print" env var in afnirc
 -----------------------------------------------------------------------------*/
 
+/* move this somewhere else later */
+
+float frac_diff( float old, float new );
+
+float frac_diff(float old, float new) {
+    float fracdiff;
+
+    if (fabs(old) + fabs(new) != 0) {
+        fracdiff = fabs(old - new) / (fabs(old) + fabs(new));
+    } else {
+        fracdiff = 0;
+    }
```

Comment:
How about perhaps:

```
    float fracdiff, denom;

    denom = fabs(old) + fabs(new);

    if (denom) {
        fracdiff = 2 * fabs(old - new) / denom;
    } else {
        fracdiff = 0.;
    }
```

... which calculates the denominator only once? A factor of 2 in the numerator makes sense to me, from the point of view of considering the denominator to be the average of old and new.